python/lsst/meas/astrom/anetAstrometry.py

```diff
@@ -0,0 +1,453 @@
+
+    See \ref meas_photocal_photocal_Example.
+
+    To investigate the \ref pipe_tasks_astrometry_Debug, put something like
+    \code{.py}
+        import lsstDebug
+        def DebugInfo(name):
+            di = lsstDebug.getInfo(name)        # N.b. lsstDebug.Info(name) would call us recursively
+            if name == "lsst.pipe.tasks.astrometry":
+                di.display = 1
+
+            return di
+
+        lsstDebug.Info = DebugInfo
+    \endcode
+    into your debug.py file and run photoCalTask.py with the \c --debug flag.
+    """
+    ConfigClass = ANetAstrometryConfig
+
+    def __init__(self, schema, **kwds):
+        """!Create the astrometric calibration task.  Most arguments are simply passed onto pipe.base.Task.
+
+        \param schema An lsst::afw::table::Schema used to create the output lsst.afw.table.SourceCatalog
+        \param **kwds keyword arguments to be passed to the lsst.pipe.base.task.Task constructor
+
+        A centroid field "centroid.distorted" (used internally during the Task's operation)
+        will be added to the schema.
+        """
+        pipeBase.Task.__init__(self, **kwds)
+
+        if schema.getVersion() == 0:
+            self.distortedName = "centroid.distorted"
+            self.centroidKey = schema.addField(self.distortedName, type="PointD",
+                                           doc="centroid distorted for astrometry solver")
+            self.centroidErrKey = schema.addField(self.distortedName + ".err", type="CovPointF",
+                                           doc="centroid distorted err for astrometry solver")
+            self.centroidFlagKey = schema.addField(self.distortedName + ".flag", type="Flag",
+                                           doc="centroid distorted flag astrometry solver")
+        else:
+            self.distortedName = "astrom_distorted"
+            self.centroidXKey = schema.addField(self.distortedName + "_x", type="D",
+                                           doc="centroid distorted for astrometry solver")
+            self.centroidYKey = schema.addField(self.distortedName + "_y", type="D",
+                                           doc="centroid distorted for astrometry solver")
+            self.centroidXErrKey = schema.addField(self.distortedName + "_xSigma", type="F",
+                                           doc="centroid distorted err for astrometry solver")
+            self.centroidYErrKey = schema.addField(self.distortedName + "_ySigma", type="F",
+                                           doc="centroid distorted err for astrometry solver")
+            self.centroidFlagKey = schema.addField(self.distortedName + "_flag", type="Flag",
+                                           doc="centroid distorted flag astrometry solver")
+            self.centroidKey = Point2DKey(self.centroidXKey, self.centroidYKey)
+            self.centroidErrKey = CovarianceMatrix2fKey((self.centroidXErrKey, self.centroidYErrKey))
```

Comment:
For future reference, the `FunctorKey` classes in afw and meas_base can help with adding all of these fields (and they can do it in a version-dependent way).  I don't think it's worth changing here, but I'll probably use that to simplify things when I sweep through after removing version 0 tables on DM-1766.